Fix #8924 - Speed up rendering of the script list

kkthxbye-code opened the pull request:
### Fixes: #8924

Loading the script list can become slow when having multiple large scripts. The reason is that django instantiates the Script class for each access in the template. This can be very slow in some cases.

The fix is simply to set `do_not_call_in_templates` = True on the callable before passing them to the template. Please let me know if there is a better option.

The template was changed slightly, as the script variable did not return the name when not instantiated. Name is now taken from Meta instead.

Commit:
Speed up rendering of the script list

jeremystretch commented on netbox/extras/views.py:
```diff
@@ -753,6 +753,8 @@ def get(self, request):
         for _scripts in scripts.values():
             for script in _scripts.values():
+                # Prevent django from instantiating the class on all accesses
+                script.do_not_call_in_templates = True
```
Seems like this should be set on the base class, rather than being set by the view.

jeremystretch commented on netbox/templates/extras/script_list.html:
```diff
@@ -34,7 +34,7 @@ <h5 class="card-header">
                   <tr>
                     <td>
-                      <a href="{% url 'extras:script' module=script.module name=class_name %}" name="script.{{ class_name }}">{{ script }}</a>
+                      <a href="{% url 'extras:script' module=script.module name=class_name %}" name="script.{{ class_name }}">{{ script.Meta.name }}</a>
```
`script.Meta.name` is not a required attribute. Instead, this should reference `script.name`. (Incidentally, `name` should default to `self.__name__`, but that's a separate issue.)

kkthxbye-code replied:
Ahh, I totally missed the `name` class attribute. I'll change that shortly.

jeremystretch commented on netbox/templates/extras/script_list.html:
```diff
@@ -34,7 +34,7 @@ <h5 class="card-header">
                   <tr>
                     <td>
-                      <a href="{% url 'extras:script' module=script.module name=class_name %}" name="script.{{ class_name }}">{{ script }}</a>
+                      <a href="{% url 'extras:script' module=script.module name=class_name %}" name="script.{{ class_name }}">{{ script.Meta.name }}</a>
```
`script.Meta.name` is not a required attribute. Instead, this should reference `script.name`.

Commit:
- Move do_not_call_in_templates to BaseScript
- Fix the name classproperty

kkthxbye-code commented:
@jeremystretch - I moved `do_not_call_in_templates`  to the base class.

I also changed the template to use script.name. I had to change it from `self.__class__.__name__` to `self.__name__`. I tested on our production system (v3.1.5) and if you omit Meta.name the name of the script will be `"type"`. I might be missing how it's supposed to work, but I think that's a bug.
